subsys/net/lib/lwm2m_client_utils/lwm2m/gnss_assistance_obj.c

```diff
@@ -204,12 +204,16 @@ static int gnss_assistance_result_code_cb(uint16_t obj_inst_id, uint16_t res_id,
 					  uint16_t data_len, bool last_block,
 					  size_t total_size)
 {
+	if (data_len != sizeof(int32_t)) {
+		return -EINVAL;
+	}
+
 	if (result_code_cb) {
-		result_code_cb((int32_t)data);
+		result_code_cb(*(int32_t *)data);
 	}
 
 	if (result != LOCATION_ASSIST_RESULT_CODE_OK) {
-		LOG_ERR("Result code %d", (int32_t)data);
+		LOG_ERR("Result code %d", *(int32_t *)data);
```

Comment:
Comparing for `result` and printing from `*data`.
I think this should be consistent. Change both to handle just the data

Reply:
True, missed that. Now fixed.